packages/webpack-cli/src/plugins/dotenv-webpack-plugin.ts

```diff
@@ -0,0 +1,196 @@
+import dotenv from "dotenv-defaults";
+import { Compiler, DefinePlugin } from "webpack";
+
+interface EnvVariables {
+  [key: string]: string;
+}
+
+interface Logger {
+  info(message: string): void;
+  warn(message: string): void;
+  error(message: string): void;
+}
+
+interface DotenvConfig {
```

Comment:
Let's implement `safe`, it is useful in some cases https://github.com/mrsteele/dotenv-webpack?tab=readme-ov-file#properties